pkg/destroy/aws/aws.go

```diff
@@ -125,12 +125,18 @@ func (o *ClusterUninstaller) Run() error {
 	tagClientNames := map[*resourcegroupstaggingapi.ResourceGroupsTaggingAPI]string{
 		tagClients[0]: o.Region,
 	}
-	if o.Region != "us-east-1" {
-		tagClient := resourcegroupstaggingapi.New(
-			awsSession, aws.NewConfig().WithRegion("us-east-1"),
-		)
+	tagConfig := aws.NewConfig()
+	switch o.Region {
+	case endpoints.CnNorth1RegionID, endpoints.CnNorthwest1RegionID:
+		tagConfig = tagConfig.WithRegion(endpoints.CnNorthwest1RegionID)
+		tagClient := resourcegroupstaggingapi.New(awsSession, tagConfig)
+		tagClients = append(tagClients, tagClient)
+		tagClientNames[tagClient] = endpoints.CnNorthwest1RegionID
+	default:
+		tagConfig = tagConfig.WithRegion(endpoints.UsEast1RegionID)
+		tagClient := resourcegroupstaggingapi.New(awsSession, tagConfig)
 		tagClients = append(tagClients, tagClient)
-		tagClientNames[tagClient] = "us-east-1"
+		tagClientNames[tagClient] = endpoints.UsEast1RegionID
 	}
```

Comment:
because we want to capture the hostedzones, when we are in us-east-1 we don't need another tag client but when we are in any other region we need to make sure we have one tag client for the regions and another tag client for us-east-1.
This allowed us to only have extra tag client when required. to prevent un necessary duplicate API calls.

so i think we need to keep that behavior. and this code changes that.

case china-north-1, china-nortwest-1:
if != china-northwest-1
   // append a tag client for china-northwest-1

default:
!= us-east-1
   // append tag client for us-east-1